core/chains/evm/headtracker/head_listener.go

```diff
@@ -28,32 +32,62 @@ var (
 	}, []string{"evmChainID"})
 )
 
-type headListener struct {
-	config           Config
-	ethClient        evmclient.Client
+type headListener[
```

Comment:
So this implements common.HeadListener right?

Could you add var _ HeadListener[...] = &evmHeadListener{}}

Reply:
thanks for the note for this! Implemented :)